pip/wheel.py

```diff
@@ -696,15 +701,42 @@ def __build_one(self, req, tempd, python_tag=None, isolate=False):
                 env['PYTHONNOUSERSITE'] = '1'
 
             try:
-                call_subprocess(wheel_args, cwd=req.setup_py_dir,
-                                extra_environ=env,
-                                show_stdout=False, spinner=spinner)
+                p = Pool()
+                response = p.apply_async(self.__build_one_inside_subprocess,
+                                         (wheel_args, req.setup_py_dir,
+                                          env)).get()
+
+                if response != 'The operation completed successfully':
+                    raise ValueError('Subprocess failed to complete')
+
+                spinner.finish()
+                return True
+            except:
+                logger.debug('Failed building wheel with PEP 517')
+
+            try:
+                call_subprocess(
+                    wheel_args,
+                    cwd=req.setup_py_dir,
+                    extra_environ=env,
+                    show_stdout=False,
+                    spinner=spinner)
                 return True
             except:
                 spinner.finish("error")
                 logger.error('Failed building wheel for %s', req.name)
                 return False
 
+    def __build_one_inside_subprocess(self, wheel_args, cwd, env):
+        os.chdir(cwd)
+        os.environ = env
+        config_settings = {"--build-option": wheel_args}
```

Comment:
One thing I'm not completely sure about is whether environ is checked as part of sys.path. Or does that need to be updated as well to env['PYTHONPATH']